### PR TITLE
rename and deprecate shelve/shelf, shelve_type/shelf_type

### PR DESCRIPTION
**Pull request by hoyer-a**
Related Issue #641 

PR to rename / deprecate `_shelve` in favor of `_shelf` and `shelve_type` in favor of `shelf_type` in functions

- high_shelve
- low_shelve
- high_shelve_cascade
- low_shelve_cascade 

and related tests & private functions.

I didn't rename any files (e.g. in tests and tests/references), should i do that too?

### CODE REVIEW

File: pyfar/dsp/filter/__init__.py [hunk 1]
```diff
@@ -19,10 +19,14 @@
...
     bell,
+    high_shelf,
     high_shelve,
     low_shelve,
+    low_shelf,
     high_shelve_cascade,
-    low_shelve_cascade
+    high_shelf_cascade,
+    low_shelve_cascade,
+    low_shelf_cascade
```
Comment (ahms5) [inline pyfar/dsp/filter/__init__.py:29]:
```suggestion
    low_shelf_cascade,
```
to make diffs easier and avoid merge conflict

File: pyfar/dsp/filter/audiofilter.py [hunk 2]
```diff
@@ -224,25 +226,132 @@ def bell(signal, center_frequency, gain, quality, bell_type='II',
         return signal_filt
 
 
-def high_shelve(signal, frequency, gain, order, shelve_type='I',
+@rename_arg({"shelve_type": "shelf_type"}, "shelve_type will be deprecated in"
+            " version 0.8.0 in favor of shelf_type")
+def high_shelve(signal, frequency, gain, order, shelf_type='I',
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:231]: I don't think we need to deprecate parameters in a deprecated function, it just confuses people. I would leave it as it is and just deprecate it and add the new one.

File: pyfar/dsp/filter/audiofilter.py [hunk 3]
```diff
@@ -279,16 +390,103 @@ def high_shelve(signal, frequency, gain, order, shelve_type='I',
...
+    output = _shelf(
+        signal, frequency, gain, order, shelf_type, sampling_rate, 'high')
 
     return output
 
 
-def low_shelve(signal, frequency, gain, order, shelve_type='I',
+@rename_arg({"shelve_type": "shelf_type"}, "shelve_type will be deprecated in"
+            " version 0.8.0 in favor of shelf_type")
+def low_shelve(signal, frequency, gain, order, shelf_type='I',
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:401]: same here

File: tests/references/filter.shelve_cascade.py [hunk 4]
```diff
@@ -1,21 +1,21 @@
-# Write cascaded shelve filters for testing.
+# Write cascaded shelf filters for testing.
 # The filter shape was inspected manually before writing the data.
```
Comment (ahms5) [inline tests/references/filter.shelve_cascade.py:2]: rename file

**Review by ahms5 — CHANGES_REQUESTED** [review]
thank you for taking care! I think we also need to deprecate it into 0.9.0, this will be released in 0.7.0 and then we need to wait 2 minor releases.

File: pyfar/dsp/filter/audiofilter.py (hunk 2 above)
Comment (hoyer-a) [reply pyfar/dsp/filter/audiofilter.py:231]: But we should properly deprecate "shelve_type" at some point or is it ok to rename the parameter if the function is renamed?

Comment (ahms5) [reply pyfar/dsp/filter/audiofilter.py:231]: Yes I think you can deprecate the old method and introduce the new one, directly with the new parameter name. If the old one has a "wrong" name it does not matter, its deprecated anymay trough the function.

**Comment by hoyer-a** [comment]
Ruff tests fail because this line is too long. Should i fix it here? It's not related to the changes i made.  
  
```
signal : :py:class:`~pyfar.classes.audio.Signal`, :py:class:`~pyfar.classes.filter.FilterSOS`
```

File: pyfar/dsp/filter/audiofilter.py [hunk 5]
```diff
@@ -349,6 +480,110 @@ def high_shelve_cascade(
...
+        >>> import matplotlib.pyplot as plt
+        >>>
+        >>> impulse = pf.signals.impulse(40e3, sampling_rate=40000)
+        >>> impulse, N, ideal = pf.dsp.filter.high_shelf_cascade(
+        >>>     impulse, 250, "lower", -60, None, 4)
+        >>>
+        >>> pf.plot.freq(ideal, c='k', ls='--', label="ideal")
+        >>> pf.plot.freq(impulse, label="actual")
+        >>> plt.legend()
+    """
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:573]:
```suggestion
    """  # noqa: E501
```
we always handle it in this way, as we know there is no other way

File: pyfar/dsp/filter/audiofilter.py [hunk 6]
```diff
@@ -461,6 +696,109 @@ def low_shelve_cascade(
...
+        >>> import matplotlib.pyplot as plt
+        >>>
+        >>> impulse = pf.signals.impulse(40e3, sampling_rate=40000)
+        >>> impulse, N, ideal = pf.dsp.filter.low_shelf_cascade(
+        >>>     impulse, 4000, "upper", -60, None, 4)
+        >>>
+        >>> pf.plot.freq(ideal, c='k', ls='--', label="ideal")
+        >>> pf.plot.freq(impulse, label="actual")
+        >>> plt.legend()
+    """
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:776]:
```suggestion
    """  # noqa: E501
```

File: pyfar/dsp/filter/audiofilter.py [hunk 7]
```diff
@@ -279,30 +283,96 @@ def high_shelve(signal, frequency, gain, order, shelve_type='I',
 blob/master/filter_design/audiofilter.py
     """
 
-    output = _shelve(
-        signal, frequency, gain, order, shelve_type, sampling_rate, 'high')
+    warnings.simplefilter('always', PyfarDeprecationWarning)
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:286]: is there a specific reason for that?

File: pyfar/dsp/filter/audiofilter.py [hunk 8]
```diff
@@ -339,8 +409,69 @@ def low_shelve(signal, frequency, gain, order, shelve_type='I',
 blob/master/filter_design/audiofilter.py
     """
 
-    output = _shelve(
-        signal, frequency, gain, order, shelve_type, sampling_rate, 'low')
+    warnings.simplefilter('always', PyfarDeprecationWarning)
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:412]: same here

File: pyfar/dsp/filter/audiofilter.py [hunk 9]
```diff
@@ -349,6 +480,110 @@ def high_shelve_cascade(
...
+    the three parameter `gain`, `slope`, and `bandwidth` must be specified,
+    while the third parameter is calculated as
+
+    ``gain = bandwidth * slope``
+
+    ``bandwidth = abs(gain/slope)``
+
+    ``slope = gain/bandwidth``
+
+    :py:func:`~pyfar.dsp.filter.high_shelve_cascade` will be deprecated in
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:496]: A general note, no idea what is more convenient, in Coordinates we always put this note at the very beginning.

File: pyfar/dsp/filter/audiofilter.py [hunk 10]
```diff
@@ -349,6 +480,110 @@ def high_shelve_cascade(
...
+        >>>
+        >>> impulse = pf.signals.impulse(40e3, sampling_rate=40000)
+        >>> impulse, N, ideal = pf.dsp.filter.high_shelf_cascade(
+        >>>     impulse, 250, "lower", -60, None, 4)
+        >>>
+        >>> pf.plot.freq(ideal, c='k', ls='--', label="ideal")
+        >>> pf.plot.freq(impulse, label="actual")
+        >>> plt.legend()
+    """
+    warnings.simplefilter('always', PyfarDeprecationWarning)
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:574]: same here

File: pyfar/dsp/filter/audiofilter.py (hunk 7 above)
Comment (ahms5) [reply pyfar/dsp/filter/audiofilter.py:286]: According to the PyfarDeprecationWarning doc, we do not need to use this.

File: pyfar/dsp/filter/audiofilter.py [hunk 11]
```diff
@@ -721,7 +1059,9 @@ def _shelve_cascade(signal, frequency, frequency_type, gain, slope, bandwidth,
         return filt.process(signal), N, ideal_response
 
 
-def _shelving_cascade_slope_parameters(gain, slope, bandwidth, shelve_type):
+@rename_arg({"shelve_type": "shelf_type"}, "shelve_type will be deprecated in"
+            " version 0.9.0 in favor of shelf_type")
+def _shelving_cascade_slope_parameters(gain, slope, bandwidth, shelf_type):
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:1064]: its a private function, so we dont need to deprecate it, we can change directly. It should just be used within pyfar and not outside

**Review by ahms5 — CHANGES_REQUESTED** [review]
Thank you, just small remarks. not sure ifwe should put the warning in the docs at the very beginning as done for Coordinates

File: pyfar/dsp/filter/audiofilter.py (hunk 7 above)
Comment (hoyer-a) [reply pyfar/dsp/filter/audiofilter.py:286]: Ah, no we don't need it, thanks. It's a leftover

File: pyfar/dsp/filter/audiofilter.py (hunk 9 above)
Comment (hoyer-a) [reply pyfar/dsp/filter/audiofilter.py:496]: Probably better to put it at the beginning

File: pyfar/dsp/filter/audiofilter.py [hunk 12]
```diff
@@ -339,8 +404,68 @@ def low_shelve(signal, frequency, gain, order, shelve_type='I',
 blob/master/filter_design/audiofilter.py
     """
 
-    output = _shelve(
-        signal, frequency, gain, order, shelve_type, sampling_rate, 'low')
+    warnings.warn(("'low_shelve' will be deprecated in pyfar 0.9.0 in favor "
+                   "of 'low_shelf'"), PyfarDeprecationWarning)
+
+    return low_shelf(signal, frequency, gain, order, shelve_type, sampling_rate)
```
Comment (f-brinkmann) [inline pyfar/dsp/filter/audiofilter.py:410]: This line is too long - not sure why the tests did not show it...

File: tests/test_deprecations.py [hunk 13]
```diff
@@ -295,3 +296,79 @@ def test_deprecations_freq_range_parameter_renaming_results():
...
+
+
+def test_deprecation_shelve_functions():
+    # test high_shelve()
+    with pytest.warns(
+        PyfarDeprecationWarning, match="'high_shelve' will be deprecated in "
+            "pyfar 0.9.0 in favor of 'high_shelf'"):
+        pfilt.high_shelve(None, 10e3, 1.5, 1, "I", sampling_rate=44100)
+
+    if version.parse(pf.__version__) >= version.parse('0.8.0'):
```
Comment (f-brinkmann) [inline tests/test_deprecations.py:308]: Should be changed here and below to:
`if version.parse(pf.__version__) >= version.parse('0.9.0'):`

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Thanks, I only found two minor things to change. Can you also merge develop back into you branch. There were changes in the docstring in https://github.com/pyfar/pyfar/pull/648 that should be included in all cascaded_she* funcionts.

**Review by f-brinkmann — APPROVED** [review]

**Review by ahms5 — APPROVED** [review]

**Pull request merged**